### PR TITLE
Fix for not specifying labels in find_overlapping_issues

### PR DESCRIPTION
**Pull request by huiwengoh**
## Changes in this PR

- _get_num_examples accepts `confident_joint` as a fallback argument.
- `confident_joint` won't be "calibrated" in `count.estimate_joint()` without provided `labels`.
- Property-based tests for `find_overlapping_classes(confident_joint=confident_joint)` added.

## New dependencies

[Hypothesis](https://hypothesis.readthedocs.io/en/latest/) is used for property-based testing. This is added as a dev dependency.

Resolves https://github.com/cleanlab/cleanlab/issues/651

### CODE REVIEW

**Comment by jwmueller** [comment]
@elisno Can consider adding your MWE as a unit test to address the missing codecov:

```
from cleanlab.dataset import find_overlapping_classes
from cleanlab.classification import CleanLearning

from sklearn.datasets import make_classification

cl = CleanLearning().fit(*make_classification())
find_overlapping_classes(confident_joint=cl.confident_joint)  # Raises the TypeError
```
